src/core/processing/qgsprocessingparametermeshdataset.cpp

```diff
@@ -113,10 +113,29 @@ QList<int> QgsProcessingParameterMeshDatasetGroups::valueAsDatasetGroup( const Q
 {
   if ( !valueIsAcceptable( value, true ) )
     return QList<int>();
-  QVariantList list = value.toList();
+
   QList<int> ret;
-  for ( const QVariant &v : list )
-    ret.append( v.toInt() );
+
+  // if invalid or empty, return only the group 0
+  if ( !value.isValid() )
+    ret << 0;
+  else
+  {
+    if ( value.type() == QVariant::List )
+    {
+      //here we can't use  QgsProcessingParameters::parameterAsInts() because this method return empry list when first value is 0...
```

Comment:
```suggestion
      //here we can't use  QgsProcessingParameters::parameterAsInts() because this method return empty list when first value is 0...
```